context-free.md

```diff
@@ -1,21 +1,32 @@
-# Context Free
+# Context Free Questions
+#### Zeng Wei Cheng
 
 ##  Who is this programming language for?
+Context-free is a external domain specific language designed specifically for artists who have little knowledge in programming. Users of this language can easily get familiar with its syntax or symbols and use it to create art work.
```

Comment:
I agree entirely with the second point you make about it being fairly easy to familiarise oneself with the syntax used by ContextFree, especially due to its linguistic nature as a computer programming language - i.e because it uses several terms which are an intrinsic part of our understanding of patterns: be it rotations or transformations, flipping or repetitions. The usage of these parameters on shapes like triangles, squares, and circles adds a sense of fluency which comes not just from the expressions but the way they can be composed together. Due to this reason, i.e. its language nature and because, as you mentioned, it is for artists and thus has a domain focus - I agree with your classification of ContextFree as a domain-specific language. However, one thing I don't entirely agree with is that one needs some knowledge of programming to use this because I believe that ContextFree's language nature helps prevent this. Even to use traditional programming concepts such as loops in ContextFree, I believe the right documentation is enough to enable one to use it even without prior programming experience.